webpack/InsightsCloudSync/InsightsCloudSyncActions.js

```diff
@@ -1,25 +1,42 @@
-import React from 'react';
 import { post } from 'foremanReact/redux/API';
 import { translate as __ } from 'foremanReact/common/I18n';
 import { insightsCloudUrl } from './InsightsCloudSyncHelpers';
-import { INSIGHTS_CLOUD_SYNC } from './InsightsCloudSyncConstants';
-import { foremanUrl } from '../ForemanRhCloudHelpers';
+import {
+  INSIGHTS_CLOUD_SYNC,
+  INSIGHTS_CLOUD_SYNC_TASK,
+} from './InsightsCloudSyncConstants';
+import { setupTaskPolling, taskRelatedToast } from '../common/ForemanTasks';
 
-export const syncInsights = () =>
-  post({
-    key: INSIGHTS_CLOUD_SYNC,
-    url: insightsCloudUrl('tasks'),
-    successToast: response => (
-      <span>
-        {__('Recommendation sync has started: ')}
-        <a
-          target="_blank"
-          rel="noopener noreferrer"
-          href={foremanUrl(`/foreman_tasks/tasks/${response.data?.task?.id}`)}
-        >
-          {__('view the task in progress')}
-        </a>
-      </span>
-    ),
-    errorToast: error => `${__('Recommendation sync has failed: ')} ${error}`,
+export const syncInsights = (fetchInsights, query) => dispatch =>
+  dispatch(
+    post({
+      key: INSIGHTS_CLOUD_SYNC,
+      url: insightsCloudUrl('tasks'),
+      handleSuccess: ({
+        data: {
+          task: { id },
+        },
+      }) => {
+        dispatch(syncInsightsStartedToast(id));
+        dispatch(setupInsightsTaskPolling(id, fetchInsights, query, dispatch));
+      },
+      errorToast: error => syncInsightsError(error),
+    })
+  );
+
+const syncInsightsError = error =>
+  `${__('Recommendation sync has failed: ')} ${error}`;
+
+const syncInsightsStartedToast = taskId =>
+  taskRelatedToast(taskId, 'info', __('Recommendation sync has started: '));
+
+const setupInsightsTaskPolling = (taskId, fetchInsights, query, dispatch) =>
+  setupTaskPolling({
+    taskId,
+    key: INSIGHTS_CLOUD_SYNC_TASK,
+    onTaskSuccess: () => {
+      fetchInsights({ query, page: 1 });
```

Comment:
looks like I get the success callback, but for some reason I see a toast with the error: 
`Could not get task details: TypeError: Cannot read property 'type' of undefined`

Reply:
I needed to change in `add_satellite_notifications` the line:
`blueprint&.notifications&.destroy_all`
 Because I got an error that notifications is undefined and then I started getting this error as well

Reply:
seem that because of the `dispatch` in https://github.com/theforeman/foreman_rh_cloud/pull/561/files#diff-062e73f73ae1ac5114f5c53151d1230ee336d3c844290d4d966c8f1236b8a26cR24
and because this `onTaskSuccess` returns nothing, eventually what's happening is `dispatch(undefined)`
action should be in a type of object with `{type: 'SOME-TYPE', payload: {}}`
I would also take a closer look on https://github.com/theforeman/foreman_rh_cloud/pull/561/files#diff-5e22a22b04f74adeb26948f4c823ba96e6fa74fffe6823468cb0d65df73c2f43R40